packages/payment/src/workflows/steps/create-payment-session.ts

```diff
@@ -0,0 +1,39 @@
+import { PaymentProviderContext } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  payment_collection_id: string
+  provider_id: string
+  amount: number
+  currency_code: string
+  context: PaymentProviderContext
+  data: Record<string, unknown>
+}
+
+export const createPaymentSessionStepId = "module:create-payment-session"
+export const createPaymentSessionStep = createStep(
+  createPaymentSessionStepId,
+  async (data: StepInput, { container }) => {
+
+    const service = container.resolve("paymentSessionService")
+
+    const created = await service.create({
```

Comment:
Here we can pass the sharedContext to the create use the same DB transaction

Reply:
I wouldn't do that, plus we still have a LRT here and it would just increase it, lets discuss tomorrow

Reply:
yes, disregard it competely. that was the reason to use workflows to begin with..
🤦🏻‍♂️

Reply:
Yea thats should suffice, and replace it with Inject manager instead. The call to the inner service will start a transaction block for it, and if we have multiple actions to do then we can use this.baseRepository_.transaction to wrap multiple calls in a single transaction which can also take a transaction if one already started but it shouldn't be the case here